netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -148,13 +149,14 @@ public void operationComplete(ChannelFuture future) throws Exception {
     @Override
     public void writeFrame(
         WritableBuffer frame, boolean endOfStream, boolean flush, int numMessages) {
+      Preconditions.checkArgument(numMessages >= 0);
       ByteBuf bytebuf = frame == null ? EMPTY_BUFFER : ((NettyWritableBuffer) frame).bytebuf();
       final int numBytes = bytebuf.readableBytes();
       if (numBytes > 0) {
         // Add the bytes to outbound flow control.
         onSendingBytes(numBytes);
         writeQueue.enqueue(
-            new SendGrpcFrameCommand(transportState(), bytebuf, endOfStream, numMessages),
+            new SendGrpcFrameCommand(transportState(), bytebuf, endOfStream),
             channel.newPromise().addListener(new ChannelFutureListener() {
               @Override
               public void operationComplete(ChannelFuture future) throws Exception {
```

Comment:
Forget to reportMessageSent in here?

Reply:
That will be added as a part of the diff to add TransportTracer to client side, right now the field is not defined yet.